eve/main.yml

```diff
@@ -1617,7 +1653,26 @@ stages:
             TF_VAR_rhsm_username: "%(secret:rhel_ci_login)s"
             TF_VAR_rhsm_password: "%(secret:rhel_ci_password)s"
       # --- Wait for all pods to be running ---
-      - ShellCommand: *wait_pods_running_ssh
+      - ShellCommand:
+          <<: *wait_pods_running_ssh
+          # NOTE: Do not use `wait_pods_running_ssh` directly as we need to
+          # have a workaround to make sure all PVs are well recreated by
+          # storage operator
+          # Check https://github.com/scality/metalk8s/issues/2726
+          # As workaround if `wait_pods_status` failed just restart salt-minion
+          # everywhere and `wait_pods_status` again
+          command: >
+            git checkout "%(prop:branch)s" &&
+            scp -F ssh_config "%(prop:builddir)s/build/eve/wait_pods_status.sh" bootstrap:/tmp/ &&
+            ssh -F ssh_config bootstrap sudo bash -c "
+            /tmp/wait_pods_status.sh --sleep-time 5 --stabilization-time 30 --status Running
+            --retry 30 || (crictl exec -it
+            \$(sudo crictl ps -q --label io.kubernetes.pod.namespace=kube-system
+            --label io.kubernetes.container.name=salt-master --state Running)
+            salt '*' cmd.run_bg 'salt-call --local service.restart salt-minion' &&
+            sleep 10 &&
+            /tmp/wait_pods_status.sh --sleep-time 5 --stabilization-time 30 --status Running
+            --retry 30)"
```

Comment:
Shouldn't we always restart salt-minion, "just in case", instead of having this ?

Reply:
Hmm right, we could, but since the logic is here now let's keep it for the moment :smile: